src/App.tsx

```diff
@@ -1,19 +1,175 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos, addTodo, toggleStatus, removeTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+
+export enum FilterStatus {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+type Title = {
+  title?: null | string
+};
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isHidden, setIsHidden] = useState(true);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [notCompleted, setNotCompleted] = useState(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Title>({ title: null });
+  const [activeTodo, setActiveTodo] = useState(true);
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const onChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTodoTitle('');
+    const addNewTodo = async (data: Omit<Todo, 'id'>) => {
+      try {
+        const todo = await addTodo(data);
+
+        setTodos((currentTodos): Todo[] | any => [...currentTodos, todo]);
+
+        setTempTodo({ title: null });
+      } catch {
+        setErrorMessage('Unable to add a todo');
+        setIsHidden(false);
+        setTempTodo({ title: null });
+      }
+    };
+
+    if (user === null) {
+      return;
+    }
+
+    if (todoTitle === '') {
+      setErrorMessage('Title can\'t be empty');
+      setIsHidden(false);
+    }
+
+    if (todoTitle !== '') {
+      setTempTodo({
+        title: todoTitle,
+      });
+      addNewTodo({
+        userId: user.id,
+        title: todoTitle,
+        completed: false,
+      });
```

Comment:
```suggestion
    if (todoTitle.trim() !== '') {
      setTempTodo({
        title: todoTitle,
      });
      addNewTodo({
        userId: user.id,
        title: todoTitle,
        completed: false,
      });
```

it's better to check todoTitle.trim() !== '', because now i can add empty title ('     ')